### PR TITLE
Add check to the mod init function so users are aware if it's run in the home directory 

### PR DESCRIPTION
**Pull request by Aaronstotle**
This adds a small function to check the location of steampipe mod init so users receive a warning if they run this in their home directory.

### CODE REVIEW

**Comment by binaek** [comment]
Hi @Aaronstotle,

Welcome to Steampipe and thank you for raising the PR.

Out of curiosity, what issue were you facing with `mod init` that you believe this change is necessary?

**Comment by Aaronstotle** [comment]
https://github.com/turbot/steampipe/issues/2562

I found this issue while browsing

**Comment by binaek** [comment]
Thank you for working on that one. 👍 

Correct me if I am wrong - but the changes tackle the part where it checks if the `init` directory is the `home directory`

Shouldn't we also add the `>5 files/folders not .sql/.sp` check as well? That will prevent users from running `mod init` in a directory which wasn't meant for it.

**Review by binaek — CHANGES_REQUESTED** [review]

**Comment by binaek** [comment]
@Aaronstotle all our exit codes are listed in `pkg/constants/exit_codes.go`

It looks like we do not have an exit code (yet) for `mod init` failures.

@kaidaguerre I think we should add an exit code for this. e.g: `ExitCodeModInitFailed` with a value in the `(60,69)` range.

cc @pskrbasu

File: cmd/mod.go [hunk 1]
```diff
@@ -223,8 +252,12 @@ func runModInitCmd(cmd *cobra.Command, args []string) {
 		}
 	}()
 	workspacePath := viper.GetString(constants.ArgModLocation)
+	// Add testing so I can run this and verify
+	home, _ := os.UserHomeDir()
+	verifyModLocation(workspacePath, home)
 	if parse.ModfileExists(workspacePath) {
 		fmt.Println("Working folder already contains a mod definition file")
+
```
[the remark this answers is not in the document]
Comment (Aaronstotle) [reply cmd/mod.go]: done

File: cmd/mod.go [hunk 2]
```diff
@@ -223,8 +252,12 @@ func runModInitCmd(cmd *cobra.Command, args []string) {
 		}
 	}()
 	workspacePath := viper.GetString(constants.ArgModLocation)
+	// Add testing so I can run this and verify
+	home, _ := os.UserHomeDir()
```
[the remark this answers is not in the document]
Comment (Aaronstotle) [reply cmd/mod.go]: thanks, moved this to within function in the updated file

**Comment by kaidaguerre** [comment]
Hey @Aaronstotle apologies for the delay in responding. 

Many thanks for the PR. I have a few comments/suggestions.


Firstly, I think `verifyModLocation` should _not_ call `os.Exit` or output a message, instead just return a boolean of whether to continue. The calling code and exit and show an appropriate message.

Secondly it's worth bearing in mind that mod initialization of a new mod happens from a couple of places:
1) the `mod init` command, as covered in this PR
2) the `mod install` command - _if there is not already a mod.sp in the current folder, one is created_

We need this check in both places. 

For the mod install command, the code which checks for the existence of a mod-file is quite low down (https://github.com/turbot/steampipe/blob/404dd35e211b1128ebe30af5a38adf8ace9ee480/pkg/modinstaller/mod_installer.go#L453)

I think the best approach to repeat the check for a mod-file existence in `runModInstallCmd`: 
```
	if !parse.ModfileExists(opts.WorkspacePath) {
		if !verifyModLocation(opts.WorkspacePath) {
			fmt.Println("Stopping mod installation")
			exitCode = tbd
			return
		}
	}
```

**Comment by kaidaguerre** [comment]
If you are still interested in making these changes and need any further input, please reach out to me

**Pull request merged**